### PR TITLE
My Jetpack: reorganizing icons - tweak icons for interstitial pages

### PR DESCRIPTION
**Pull request by retrofox**
This PR reorganizes the icons in order to use and tweak their size easily. It's needed to adjust the size of the icons for the interstitial pages.

Fixes #

#### Changes proposed in this Pull Request:

* My Jetpack: re organize icons - tweak icons for interstitial pages

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Go to My Jetpack.
* Confirm visually that icons are properly rendered in every card.
image
* Confirm the icon for the interstitial page, for instance `https://your-testing-site/wp-admin/admin.php?page=my-jetpack#/add-search`

Screen Shot 2022-02-10 at 5 17 02 PM

**Storybook**

* Run storybook

```bash
cd ./jetpack/projects/js-packages/storybook
```

```bash
> pnpm install & pnpm storybook:dev
```

Take a look at the Icons section

image

### CODE REVIEW

**Comment by oskosk** [comment]
Just rebased this PR in db2fb1a8039a0c4ddcf243102d9a169fb617b5f5

I'm confused. Seems like this PR is introducing icons for Search and Boost interstitial in spite of the clarification from #22776


> If we're highlighting a single product (Search, CRM, VideoPress) we shouldn't show it.

**Comment by retrofox** [comment]
We need icons not only for the interstitial pages but for the dashboard. It moves the icons into icons folder. After that we can use them wherever it will required

**Comment by oskosk** [comment]
>We need icons not only for the interstitial pages but for the dashboard. It moves the icons into icons folder. After that we can use them wherever it will required


yes, but this PR is also doing this 
image

Which adds the icons to the interstitial pages for search and boost if I'm not mistaken.   That's what Filipe said we only needed when showing product plans on the left and bundles on the right

**Comment by retrofox** [comment]
Yeah. I'm away about that. Probably something that was there because a bad rebase or something like that. I'm afk now but going to update ASAP

**Comment by retrofox** [comment]
> Yeah. I'm away about that. Probably something that was there because a bad rebase or something like that. I'm afk now but going to update ASAP

Addressed here https://github.com/Automattic/jetpack/pull/22799/commits/71623a9837ae09df0a79dcf826ff7ad8c02bc2e5

**Review by oskosk — DISMISSED** [review]

File: projects/packages/my-jetpack/_inc/components/product-cards-section/backup-card.jsx [hunk 1]
```diff
@@ -9,18 +9,8 @@ import PropTypes from 'prop-types';
...
-	<svg width="24" height="24" fill="none" xmlns="http://www.w3.org/2000/svg">
-		<path
-			fillRule="evenodd"
-			clipRule="evenodd"
-			d="m15.82 11.373.013-1.277v-.03c0-1.48-1.352-2.899-3.3-2.899-1.627 0-2.87 1.014-3.205 2.207l-.32 1.143-1.186-.048a2.192 2.192 0 0 0-.089-.002c-1.19 0-2.233 1.008-2.233 2.35 0 1.34 1.04 2.348 2.23 2.35H16.8c.895 0 1.7-.762 1.7-1.8 0-.926-.649-1.643-1.423-1.776l-1.258-.218ZM7.883 8.97l-.15-.003C5.67 8.967 4 10.69 4 12.817c0 2.126 1.671 3.85 3.733 3.85H16.8c1.767 0 3.2-1.478 3.2-3.3 0-1.635-1.154-2.993-2.667-3.255v-.045c0-2.43-2.149-4.4-4.8-4.4-2.237 0-4.118 1.404-4.65 3.303Z"
-			fill="#1E1E1E"
-		/>
-	</svg>
-);
+import useMyJetpackNavigate from '../../hooks/use-my-
```
Comment (oskosk) [inline projects/packages/my-jetpack/_inc/components/product-cards-section/backup-card.jsx:12]: This is broken here.

File: projects/packages/my-jetpack/_inc/components/icons/stories/index.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+					<div className={ styles[ 'icons-container' ] }>
+						{ Object.keys( allIcons ).map( key => {
+							const Icon = allIcons[ key ];
+							return (
+								<div
+									className={ `${ styles[ 'icon-wrapper' ] } ${ styles[ size.label ] }` }
+									key={ key }
+								>
+									<Icon size={ size.value } />
+									<span>{ Icon.displayName.replace( /icon/gi, '' ) }</span>
```
Comment (oskosk) [inline projects/packages/my-jetpack/_inc/components/icons/stories/index.jsx:53]: I'm not sure I understand why are we doing this. Looks super hacky

**Review by oskosk — CHANGES_REQUESTED** [review]

**Comment by oskosk** [comment]
Not sure what happened here. It was working last night when I rebased it. I see that my rebase commit (https://github.com/Automattic/jetpack/commit/db2fb1a8039a0c4ddcf243102d9a169fb617b5f5)  is gone now and some code is broken. See https://github.com/Automattic/jetpack/pull/22799#discussion_r804555095

File: projects/packages/my-jetpack/_inc/components/icons/stories/index.jsx (hunk 2 above)
Comment (retrofox) [reply projects/packages/my-jetpack/_inc/components/icons/stories/index.jsx:53]: it's the stories. It just iterates along all icons to render them. I don't think we need to concern care too much about it.

**Comment by retrofox** [comment]
> 28bb79e

broken code is addressed here https://github.com/Automattic/jetpack/pull/22799#:~:text=fix%20horrible%20issue-,28bb79e,-Merge%20state

**Comment by retrofox** [comment]
> I see that my rebase commit ([db2fb1a](https://github.com/Automattic/jetpack/commit/db2fb1a8039a0c4ddcf243102d9a169fb617b5f5)) is gone

I've rebased with master too. Probably it got discarded because of that. Beyond this, I don't see this PR is removing or changing unrelated code.

File: projects/packages/my-jetpack/_inc/components/product-cards-section/backup-card.jsx (hunk 1 above)
Comment (retrofox) [reply projects/packages/my-jetpack/_inc/components/product-cards-section/backup-card.jsx]: https://github.com/Automattic/jetpack/commit/28bb79e083cd62dca1ebee5db566927449807b2b

File: projects/packages/my-jetpack/_inc/components/icons/stories/index.jsx (hunk 2 above)
Comment (oskosk) [reply projects/packages/my-jetpack/_inc/components/icons/stories/index.jsx:53]: 👍 I meant the replace thing...

**Review by oskosk — APPROVED** [review]
LGTM!

File: projects/packages/my-jetpack/_inc/components/icons/stories/index.jsx (hunk 2 above)
Comment (retrofox) [reply projects/packages/my-jetpack/_inc/components/icons/stories/index.jsx:53]: ohhh, yeah. Well, I did it to get the Icon name from the product class. We could add a `titleComponentMap`, or change the displayName ... but I don’t know 🤷‍♂️

**Pull request merged**